src/01_getting_started/02_why_async.md

```diff
@@ -1,45 +1,121 @@
-different webpages at the same time, you would spread the work across two
-different threads, like this:
+Asynchronous programming, or async for short, is a _concurrent programming model_
+supported by an increasing number of programming languages.
+It lets you run a large number of concurrent
+tasks on a small number of OS threads, while preserving much of the
+look and feel of ordinary synchronous programming, through the
+`async/await` syntax.
+
+## Async vs other concurrency models
+
+Since the inception of concurrent programming, several models have been
+proposed with the intention of making concurrency simple and performant.
+Although details vary between different languages and frameworks,
+a brief overview of the main categories can help you understand how
+asynchronous programming fits within the field of concurrent programming:
+
+- **OS threads** don't require any changes to the programming model,
+  which makes them easier to
+  use than async. However, they come with a large performance overhead.
+  Thread pools can mitigate some of these costs, but not enough to support
+  massive IO-bound workloads.
+- **Event-driven programming**, in conjunction with _callbacks_, can be very
+  performant, but tends to result in a verbose, "non-linear" control flow.
+  Data flow and error propagation is often hard to follow.
+- **Coroutines**, like threads, don't require changes to the programming model,
+  which makes them easy to use. Like async, they can also support a large
+  number of tasks. However, they abstract away low-level details that
+  are important for systems programming and custom runtime implementors.
+- **The actor model** divides all concurrent computation into units called
+  actors, which communicate through fallible message passing, much like
+  in distributed systems. It can be efficiently implemented, but it leaves
+  many practical issues unanswered, such as flow control and retry logic.
+
+In short, asynchronous programming allows highly performant implementations
+that are suitable for low-level languages like Rust, while providing
+most of the ergonomic benefits of threads and coroutines.
+
+## Async in Rust vs other languages
+
+Although asynchronous programming is supported in many languages, some
+details vary across implementations. Rust's implementation of async
+differs from most languages in a few ways:
+
+- **Futures are inert** in Rust and make progress only when polled. Dropping a
+  future stops it from making further progress.
+- **Async is zero-cost** in Rust, and does not require heap allocations or
+  dynamic dispatch.
+  As such, you can use async Rust in constrained environments, such as embedded
+  systems.
+- **No built-in runtime** is provided by Rust. Instead, runtimes are provided by
+  community maintained crates.
+- **Both single- and multithreaded** runtimes are available in Rust, which have
+  different strengths and weaknesses.
+
+## Async vs threads in Rust
```

Comment:
This overlaps with the section comparing async to other concurrency models; it may be worth combining them. I would also explicitly explain why async is more performant than threads. Lastly, I know you mention the single and multi threaded runtimes above, and that async is compatible with threads at the end, but this section really does read like you have to choose one or the other.

Reply:
Re overlap: To some extent yes, but we have to cover the obvious options in Rust too in order to provide actionable decision criteria for prospective users. Open to making the distinction more clear.

Re why more performant: Added a sentence to this section - but not going into detail which is complex. The gist is that a small amount of threads can handle a large amount of tasks.

Re thread-async interop: I think it's ok, because it will generally be a binary choice. An unrelated comment softened the wording a bit too.